crates/http/src/error.rs

```diff
@@ -0,0 +1,27 @@
+use axum::{
+    http::StatusCode,
+    response::{IntoResponse, Response},
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("serialization error: {0}")]
+    Serde(#[from] serde_json::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+impl serde::Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_ref())
+    }
+}
+
+impl IntoResponse for Error {
+    fn into_response(self) -> Response {
+        (StatusCode::INTERNAL_SERVER_ERROR, "UNHANDLED_CLIENT_ERROR").into_response()
```

Comment:
I believe this should also work, no?

```suggestion
        (StatusCode::INTERNAL_SERVER_ERROR, self.to_string()).into_response()
```

Reply:
also, maybe work for later, but internal server error may not be correct.
e.g. if this was a serde_json error when parsing request data, it could mean the input was wrong, in which case this would be Unprocessable Entity.

I'm fine with keeping this as the only case though, if it makes it simpler

Reply:
I'll add a note to revisit this later on as it also might need to change for the `/iron/` endpoints